modules/http4s-munit-testcontainers/src/main/scala/munit/HttpFromContainerSuite.scala

```diff
@@ -132,9 +139,39 @@ abstract class HttpFromContainerSuite
                 case Left(t) => IO.raiseError(t)
               }
             }
+          runInParallel.fold(sequentialExecution(run)) {
+            case false => sequentialExecution(run)
+            case true  => parallelExecution(run)
+          }
         }
       }
 
+    /** Allows to run the same test several times sequencially */
+    def repeat(times: Int) = copy(repetitions = times.some)
+
+    /** Allows to run the tests in paralell */
+    def parallel = copy(runInParallel = true.some)
+
+    def parallel(maxParallel: Int) = copy(runInParallel = true.some, maxParallel = maxParallel.some)
+
+    /** Force the test to be executed just once */
+    def doNotRepeat = copy(repetitions = None)
+
+    private def sequentialExecution(run: IO[Any]) = Stream
+      .emits(1 to repetitions.getOrElse(1))
+      .covary[IO]
+      .evalMap(_ => run)
+      .compile
+      .drain
+      .unsafeRunSync()
+
+    private def parallelExecution(run: IO[Any]) = Stream
+      .emits(1 to repetitions.getOrElse(1))
+      .covary[IO]
+      .parEvalMapUnordered(maxParallel.getOrElse(1))(_ => run)
```

Comment:
**Suggestion:** You don't need to separate into `sequential` and `execution` since calling `parEvalMapUnordered` with `maxConcurrent = 1` is essentially calling `evalMap`. So you can just do:

```scala
Stream
  .emits(1 to repetitions.getOrElse(1))
  .covary[IO]
  .parEvalMapUnordered(maxParallel.getOrElse(1)) { _ =>
    client
      .run(request.withUri(uri))
      .use { response =>
        ...
        ...
    }
  }
  .compile
  .drain
  .unsafeRunSync()
```

Also, if we do this, we no longer need the `runInParallel` class argument.

Reply:
See [`4384ecc` (#40)](https://github.com/alejandrohdezma/http4s-munit/pull/40/commits/4384ecccc1a6caea8fd4b357723042e76004d4a9)